### DISCUSSION TITLE
Allow adding a unit hint when exporting variables

### DISCUSSION THREAD

**Discussion by mieldepoche** [godotengine/godot-proposals#4839] [Scripting]
In godot 4 some exports have units hints:
![image](https://user-images.githubusercontent.com/71937306/178089636-ffa8912b-c538-4ea1-a966-46f68a10ecca.png)

AFAIK it's not possible to specify your own unit hints for exports yet.

What about `@export("unit hint") var foo` ?
The hint would be shared among vector components if needed.

Examples:
```gdscript
@export("m/s") var speed := 10.0
@export_range(1, 60, 1, "", "s") var lifetime := 10.0
```

**Comment by Calinou** [comment]
I think you can already specify a custom hint string with annotations somehow. I don't remember the syntax though; https://github.com/godotengine/godot/pull/62713 may have documented this.

**Reply by mieldepoche** [reply]
Yeah, I made a proposal but was closed for this reason.
Syntax is:

```gdscript
# Shows `unit` as a suffix. Only works for @export_range
@export_range(0, 20, 1, "suffix:unit") var x
```

**Reply by dalexeev** [reply]
@mieldepoche Is your issue resolved? Note that in 4.3 we also added `@export_custom`, which allows you to provide arbitrary hint, hint string, and usage flags.
